app/components/Form/EditorField.js

```diff
@@ -10,31 +16,47 @@ type Props = {
   className?: string,
   input: any,
   meta: any,
-  name: string
+  name: string,
+  uploadFile: (file: Blob) => Promise<*>
 };
 
-class NoSSRError {
-  error: Error;
-  constructor(msg) {
-    this.error = new Error(msg);
-  }
-}
+const mapDispatchToProps = dispatch => {
+  return {
+    uploadFile: async file => {
+      const response = await dispatch(uploadFile({ file, isPublic: true }));
+      return { fileKey: response.meta.fileKey };
+    }
+  };
+};
 
-function EditorField({ className, name, ...props }: Props) {
-  if (!__CLIENT__) {
-    throw new NoSSRError('Cannot SSR editor');
-  }
+const EditorFieldComponent = ({
+  className,
+  name,
+  uploadFile,
+  ...props
+}: Props) => {
   return (
     <div name={name}>
       <Editor
         className={cx(styles.input, className)}
         {...props}
         {...props.input}
         {...props.meta}
+        imageUpload={uploadFile}
       />
     </div>
   );
-}
+};
+
+const EditorField = connect(
+  null,
+  mapDispatchToProps
+)(EditorFieldComponent);
+
+// $FlowFixMe
+EditorField.Field = connect(
+  null,
+  mapDispatchToProps
```

Comment:
Also, this seems a bit fishy. Connecting inside a random component.

Reply:
It's fishy yes, but I'm pretty sure its the right way to do it, as otherwise you would have to connect the route, which I think is just more confusing, as you would have to do this everywhere the editor is used and pass the props through a lot of components, many of which are abstract or HOC.